### PR TITLE
Add device automation triggers for Icasa wall-mount remotes

### PR DESCRIPTION
**Pull request by Emrvb**
This pull request supersedes #1039 

Quirk for icasa wall mounted zigbee remotes ICZB-KPD12, ICZB-KPD14S and ICZB-KPD18S. Remotes are mentioned here https://github.com/zigpy/zha-device-handlers/issues/503

ICZB-KPD12: This remote has ON and OFF buttons (hold for dimming up/down)
ICZB-KPD14S: This remote has ON and OFF buttons (hold for dimming up/down) and 2 SCENE buttons (hold for storing, push to recall).
ICZB-KPD18S: This remote has ON and OFF buttons (hold for dimming up/down) and 6 SCENE buttons (hold for storing, push to recall).

## Checklist

- [ ] The changes are tested and work correctly
- [ ] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by Emrvb** [comment]
If somebody cares to test this, that would be great. I've cleaned up this quirk a little while back, but couldn't be bothered to test it. Just did some updates, had to restart HA and the custom quirk came alive. _Appears_ to work as intended.

**Comment by Emrvb** [comment]
And if I understand the CI logs correctly a little refactoring is required, will try to do that soon.

**Comment by TheJulianJES** [comment]
Thanks for the PR!
Since there's no "real code" (just defining device triggers), we don't really need new tests for this device.
However, CI complains about some issues where triggers aren't unique and that `ARGS` need to be migrated to `PARAMS`.

**Comment by Emrvb** [comment]
@TheJulianJES Yeah, that was fixed already but my commits are a bit messy on this one, so I accidentally reverted that fix when reverting something else.

My apologies to anyone who was waiting for this quirk, but better late than never.

File: zhaquirks/icasa/iczb_kpd12.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    OUTPUT_CLUSTERS,
+    PARAMS,
+    PROFILE_ID,
+    SHORT_PRESS,
+    TURN_OFF,
+    TURN_ON,
+)
+
+COMMAND_STORE = "store"
+COMMAND_RECALL = "recall"
```
Comment (TheJulianJES) [inline zhaquirks/icasa/iczb_kpd12.py]: Do you want to move this to https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/const.py?

**Review by TheJulianJES — COMMENTED** [review]
Thanks, looks good! One minor note about adding the constants used everywhere to the global constants file.

**Comment by Emrvb** [comment]
@TheJulianJES Sure, np.

**Review by TheJulianJES — APPROVED** [review]
LGTM, thanks!
Just to confirm, triggers all work as expected? 😄

**Comment by javicalle** [comment]
I'm not really sure but if there's no changes in the `replacement` part I believe that you don't need to include it.
Can you test your quirk only with the signature and the device_triggers part?

**Comment by Emrvb** [comment]
@TheJulianJES To be brutally honest, I haven't tested the last 2 commits and I only own the ICZB-KPD18S.

@javicalle I suspect you're right, the replacement part is there because of an experiment with a custom cluster, but I never got any feedback on whether I should do that (#1041), so I ripped that out for this pull request.

Do you happen to know if I should remove the replacement variable entirely or must it be defined with some empty value?

**Comment by javicalle** [comment]
> Do you happen to know if I should remove the replacement variable entirely or must it be defined with some empty value?

Try to remove completely all the `replacement` definition.

**Comment by Emrvb** [comment]
@javicalle

I get failed tests when I remove the replacement definition. It is required when inheriting from CustomDevice. I couldn't find a relevant example to see if I should inherit from something else.

The cleanest quirk I came across is https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/sonoff/button.py which has a replacement that's identical to the signature.

Feel free to merge.

**Pull request merged**